crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -1337,7 +1337,11 @@ mod private {
                                 .expect("should have been a valid position of an item");
                         }
                         EventLocation::InStore => {
-                            todo!()
+                            self.send_updates_to_store(vec![Update::ReplaceItem {
```

Comment:
Out of curiosity, why to include the fix to your own previous commit as a separate commit?

Reply:
I wanted to go step-by-step: first the refactoring, then the fix (that's only possible because of the refactoring). Glad to merge these 2 patches in a single one when review will be finished.